main.go

```diff
@@ -485,18 +486,18 @@ func abortUpload(num int) error {
 	} else if !os.IsNotExist(err) {
 		return err
 	}
-
-	tmpsig := path.Join(directory, "tmp", strconv.Itoa(num)+".asc")
-	_, err = os.Stat(tmpsig)
-	if err == nil {
-		err = os.Remove(tmpsig)
-		if err != nil {
+	tmpsig := path.Join(directory, "tmp", strconv.Itoa(num) + ".asc")
+	if _, err := os.Stat(tmpsig); err == nil {
+		_, err = os.Stat(tmpsig)
+		if err == nil {
```

Comment:
This looks like you call `os.Stat` twice in a row on `tmpsig`, checking that err is nil both times. Is this necessary?